src/posts/2021-05-03/index.md

```diff
@@ -0,0 +1,659 @@
+  enabled. This led to a few discoveries: Version 5.5 of the Linux kernel
+  performed sensibly better than the previous ones, but version 5.10
+  reintroduced some overhead. The post, nearly a scientific report, contains
+  more details on the methodology and on the results.
+
+<!--2021-02-25-->
+* [_Exploring a New Detection Evasion Technique on Linux_](https://codemuch.tech/2021/02/25/exploring-linux-evasion/),
+  <span style="white-space: nowrap;">from Alan Cao.</span>  
+  One of the use case for application monitoring with eBPF is to ensure the
+  security of the system and to detect any mischievous process. Over the years,
+  some malwares have been found to implement countermeasures to evade
+  detection, and behave differently depending on whether they are under
+  observation or not. Can this be applied to eBPF monitoring? Yes, explains
+  this post. eBPF events can be logged to systemd, and if a process is able to
+  access these logs, it can check whether an eBPF program has been loaded at
+  the same time as itself, which would likely indicate a tentative to monitor.
+  Limitations and possible mitigations are included in the post.
+
+<!--2021-04-01-->
+* [_Examining Problematic Memory in C/C++ Applications with BPF, perf, and Memcheck_](https://doordash.engineering/2021/04/01/examining-problematic-memory-with-bpf-perf-and-memcheck/),
+  <span style="white-space: nowrap;">from Filip Busic.</span>  
+  This long post goes way beyond what one usually finds in introductions and
+  tutorials. It explains how to simply trace memory leaks in applications with
+  Memcheck, and then turns to perf and eBPF. But the author also strives to
+  provide all the technical background to understand what is going on: Stack
+  unwinding and how to do it, Linux's memory model, tracing event sources,
+  introduction to perf and eBPF, installation steps, are as many sections to
+  read and to learn from. At last, a variety of example use cases, often
+  involving flame graphs, show how to help troubleshoot memory issues.
+
+<!--2021-04-09-->
+* [_Extending systemd Security Features with eBPF_](https://kinvolk.io/blog/2021/04/extending-systemd-security-features-with-ebpf/),
+  <span style="white-space: nowrap;">from Mauricio Vásquez Bernal.</span>  
+  Two new systemd properties implemented through eBPF programs are in
+  development, leveraging the oncoming support for eBPF programs written in C
+  (rather than bytecode only) for systemd.
+
+  The first one restricts the file system types that processes in
+  a systemd unit can access. Setting `RestrictFileSystems=ext4`, for example,
+  will prevent the processes to interact with the tmpfs, thanks to a program
+  attached to the eBPF LSM (_Linux Security Module_) and running on all
+  attempts to open files.
+
+  The program checks for the presence of the magic number associated to the
+  file system in a dedicated eBPF map. The second property,
```

Comment:
```suggestion
  The program checks for the presence of the magic number associated to the
  file system in a dedicated eBPF map. 
  
  The second property,
```

What about moving the "The program checks for the presence of the magic number associated to the  file system in a dedicated eBPF map" sentence to the previous paragraph?

Reply:
Right, that was my initial intention, but it looks like I didn't break the line on the correct sentence. Good catch, and thanks for the review!